pxf/pxf-json/src/test/java/org/apache/pxf/hawq/plugins/json/PxfUnit.java

```diff
@@ -0,0 +1,666 @@
+	protected void setup(Path input) throws Exception {
+
+		if (getFragmenterClass() == null) {
+			throw new IllegalArgumentException("getFragmenterClass() must be overwritten to return a non-null object");
+		}
+
+		if (getReadAccessorClass() == null) {
+			throw new IllegalArgumentException("getReadAccessorClass() must be overwritten to return a non-null object");
+		}
+
+		if (getReadResolverClass() == null) {
+			throw new IllegalArgumentException("getReadResolverClass() must be overwritten to return a non-null object");
+		}
+
+		Map<String, String> paramsMap = new HashMap<String, String>();
+
+		// 2.1.0 Properties
+		// HDMetaData parameters
+		paramsMap.put("X-GP-ALIGNMENT", "what");
+		paramsMap.put("X-GP-SEGMENT-ID", "1");
+		paramsMap.put("X-GP-HAS-FILTER", "0");
+		paramsMap.put("X-GP-SEGMENT-COUNT", "1");
+		paramsMap.put("X-GP-FRAGMENTER", getFragmenterClass().getName());
+		paramsMap.put("X-GP-FORMAT", "GPDBWritable");
+		paramsMap.put("X-GP-URL-HOST", "localhost");
+		paramsMap.put("X-GP-URL-PORT", "50070");
+
+		paramsMap.put("X-GP-DATA-DIR", input.toString());
+
+		List<Pair<String, DataType>> params = getColumnDefinitions();
+		paramsMap.put("X-GP-ATTRS", Integer.toString(params.size()));
+		for (int i = 0; i < params.size(); ++i) {
+			paramsMap.put("X-GP-ATTR-NAME" + i, params.get(i).first);
+			paramsMap.put("X-GP-ATTR-TYPENAME" + i, params.get(i).second.name());
+			paramsMap.put("X-GP-ATTR-TYPECODE" + i, Integer.toString(params.get(i).second.getOID()));
+		}
+
+		// HDFSMetaData properties
+		paramsMap.put("X-GP-ACCESSOR", getReadAccessorClass().getName());
+		paramsMap.put("X-GP-RESOLVER", getReadResolverClass().getName());
+
+		if (getExtraParams() != null) {
+			for (Pair<String, String> param : getExtraParams()) {
+				paramsMap.put("X-GP-" + param.first, param.second);
+			}
+		}
+
+		LocalInputData fragmentInputData = new LocalInputData(paramsMap);
+
+		List<Fragment> fragments = getFragmenter(fragmentInputData).getFragments();
+
+		FragmentsResponse fragmentsResponse = FragmentsResponseFormatter.formatResponse(fragments, input.toString());
+
+		ByteArrayOutputStream baos = new ByteArrayOutputStream();
+		fragmentsResponse.write(baos);
+
+		String jsonOutput = baos.toString();
+
+		inputs = new ArrayList<InputData>();
+
+		JsonNode node = decodeLineToJsonNode(jsonOutput);
+
+		JsonNode fragmentsArray = node.get("PXFFragments");
+		int i = 0;
+		Iterator<JsonNode> iter = fragmentsArray.getElements();
+		while (iter.hasNext()) {
+			JsonNode fragNode = iter.next();
+			String sourceData = fragNode.get("sourceName").getTextValue();
+			if (!sourceData.startsWith("/")) {
+				sourceData = "/" + sourceData;
+			}
+			paramsMap.put("X-GP-DATA-DIR", sourceData);
+			paramsMap.put("X-GP-FRAGMENT-METADATA", fragNode.get("metadata").getTextValue());
+			paramsMap.put("X-GP-DATA-FRAGMENT", Integer.toString(i++));
+			inputs.add(new LocalInputData(paramsMap));
+		}
+	}
+
+	private JsonNode decodeLineToJsonNode(String line) {
```

Comment:
The PxfUnit should go in its own standalone project and be used only for testing. So better tackle any code duplications in PxfUnit once it is factored out from the pxf-json.